### PR TITLE
Be more intentional about running certain tasks in a background queue

### PR DESCRIPTION
**Pull request by czue**
Renames the `doc_deletion_queue` to `background_queue` and starts moving over anything that isn't time sensitive to that queue.

Also increases the concurrency by 2 to cover the increased load.

### CODE REVIEW

File: corehq/apps/users/tasks.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def bulk_upload_async(domain, user_specs, group_specs, location_specs):
         'messages': results
     }
 
-@task(rate_limit=2, queue='doc_deletion_queue')  # limit this to two bulk saves a second so cloudant has time to reindex
+@task(rate_limit=2, queue='background_queue')  # limit this to two bulk saves a second so cloudant has time to reindex
```
Comment (dimagimon) [inline corehq/apps/users/tasks.py]: E302 expected 2 blank lines, found 1

Comment (dimagimon) [inline corehq/apps/users/tasks.py]: E501 line too long (118 > 115 characters)

File: fab/services/templates/supervisor_celery_background_queue.conf [hunk 2]
```diff
@@ -1,6 +1,6 @@
-[program:{{ project }}-{{ environment }}-celery_doc_deletion_queue]
+[program:{{ project }}-{{ environment }}-celery_background_queue]
 environment={{ supervisor_env_vars }}
-command={{ new_relic_command }}{{ virtualenv_root }}/bin/python {{ code_root }}/manage.py celery worker --queues=doc_deletion_queue --events --loglevel=INFO --hostname={{ host_string }}_doc_deletion_queue --maxtasksperchild=50 --concurrency={{ celery_params.doc_deletion_queue.concurrency }} -Ofair
+command={{ new_relic_command }}{{ virtualenv_root }}/bin/python {{ code_root }}/manage.py celery worker --queues=background_queue --events --loglevel=INFO --hostname={{ host_string }}_doc_deletion_queue --maxtasksperchild=5 --concurrency={{ celery_params.doc_deletion_queue.concurrency }} -Ofair
```
Comment (gcapalbo) [inline fab/services/templates/supervisor_celery_background_queue.conf]: doc_deletion_queue is still referenced in the hostname and concurrency params here

File: custom/m4change/tasks.py [hunk 3]
```diff
@@ -47,7 +48,8 @@ def generate_fixtures_for_domain(domain, db, data_source):
                 FixtureReportResult.save_result(domain, location_id, date[0].date(), date[1].date(),
                                                 report_slug, rows, name)
 
-@periodic_task(run_every=crontab(hour="*", minute="*/30", day_of_week="*"), queue=getattr(settings, "CELERY_PERIODIC_QUEUE", "celery"))
+@periodic_task(run_every=crontab(hour="*", minute="*/30", day_of_week="*"),
```
Comment (dimagimon) [inline custom/m4change/tasks.py]: E302 expected 2 blank lines, found 1

File: fab/services/templates/supervisor_celery_background_queue.conf (hunk 2 above)
Comment (czue) [inline fab/services/templates/supervisor_celery_background_queue.conf]: good call. https://github.com/dimagi/commcare-hq/commit/4636d281ce7705749b84fdf56d63bfbd6ae17b92

**Comment by gcapalbo** [comment]
This looks good to me, does any QA have to happen on this?

Also, I noticed that some periodic tasks have been moved from the periodic queue to this new queue, and the new queue mixes in some long-running tasks, so we should just be aware that any periodic tasks that get executed on the new queue may appear to be slightly delayed in times of higher load.

**Comment by czue** [comment]
I'm just going to deploy to staging and make sure the supervisor stuff works fine. then i think is good to merge.

Makes sense about the delays. I pretty intentionally chose things that would not be time sensitive so I think that's ok.

**Comment by czue** [comment]
@gcapalbo supervisor looks good

**Comment by NoahCarnahan** [comment]
@czue @gcapalbo
Hey guys,
How do you normally run celery locally? I couldn't get any UCR data sources to rebuild after this update, and I eventually discovered it was because the worker I launch locally was only consuming from the default queue (named "celery"). I've been using
```./manage.py celery worker --loglevel=info --statedb=celery.db```
 to start the worker. Looks like I can fix the issue by passing `-Q celery,background_queue` to that command. Is there a way to launch a celery worker that I should know about so that it will automatically consume from all queues? If not, seems like we should probably use the [`CELERY_QUEUES`](http://celery.readthedocs.org/en/latest/configuration.html#celery-queues) setting in `settings.py` or `localsettings.py` to help keep dev environments up to date.

**Comment by czue** [comment]
@NoahCarnahan I typically just use `CELERY_ALWAYS_EAGER = True` in my localsettings which mostly means you rarely have to run celery (unless you're testing task-related behavior like progress bars). fyi @dimagi/team-commcare-hq 

http://celery.readthedocs.org/en/latest/configuration.html#celery-always-eager

**Comment by NoahCarnahan** [comment]
@czue Nice, didn't know about that setting. I'll give it a try.

**Pull request merged**